manager_janitor.go

```diff
@@ -468,16 +594,24 @@ func filterFeedable(addFeeds [][]*PIndex) (rv [][]*PIndex) {
 	return rv
 }
 
+const (
+	PHASE_CHANGE     = "phaseChange"
+	HIBERNATE_PINDEX = "hibernatePindex"
+)
+
 func classifyAddRemoveRestartPIndexes(mgr *Manager, addPlanPIndexes []*PlanPIndex,
 	removePIndexes []*PIndex) (planPIndexesToAdd []*PlanPIndex,
-	pindexesToRemove []*PIndex, pindexesToRestart []*pindexRestartReq) {
+	pindexesToRemove []*PIndex, pindexesToRestart []*pindexRestartReq,
+	pindexesToHibernate, pindexesToActivate []*pindexRestartReq) {
 	// if there are no pindexes to be removed as per planner,
 	// then there won't be anything to restart as well.
 	if len(removePIndexes) == 0 {
-		return addPlanPIndexes, nil, nil
+		return addPlanPIndexes, nil, nil, nil, nil
 	}
 	pindexesToRestart = make([]*pindexRestartReq, 0)
 	pindexesToRemove = make([]*PIndex, 0)
+	pindexesToHibernate = make([]*pindexRestartReq, 0)
+	pindexesToActivate = make([]*pindexRestartReq, 0)
```

Comment:
pindexesToHibernate and pindexesToActivate could be combined with an additional request type field to differentiate between different types of requests and a single work loop could process them later on.